go/serialization.go

```diff
@@ -561,3 +588,38 @@ func SerializeSCTList(scts []SignedCertificateTimestamp) ([]byte, error) {
 	}
 	return asn1.Marshal(buf.Bytes()) // transform to Octet String
 }
+
+// SerializeMerkleTreeLeaf writes MerkleTreeLeaf to Writer.
+func SerializeMerkleTreeLeaf(w io.Writer, m *MerkleTreeLeaf) error {
+	if m.Version != V1 {
+		return fmt.Errorf("unknown Version %d", m.Version)
+	}
+	if err := binary.Write(w, binary.BigEndian, m.Version); err != nil {
+		return err
+	}
+	if m.LeafType != TimestampedEntryLeafType {
+		return fmt.Errorf("unknown LeafType %d", m.LeafType)
+	}
+	if err := binary.Write(w, binary.BigEndian, m.LeafType); err != nil {
+		return err
+	}
+	if err := SerializeTimestampedEntry(w, &m.TimestampedEntry); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SerializeV1SCTMerkleTreeLeafX509 write the merkle tree leaf for cert
+// to Writer.
+func SerializeV1SCTMerkleTreeLeafX509(w io.Writer, timestamp uint64, cert ASN1Cert) error {
```

Comment:
Nit: naming: The X509 at the end is redundant. There'll ever only be one type of MerkleTreeLeaf (we've removed that level of indirection in 6962-bis because we found it not to be necessary), so it's not useful to designate the type of tree leaf in the function name.